update dependencies to latest stable versions

123Haynes opened the pull request:
This PR simply updates the used dependencies to their latest stable versions. 😃 

Note that I removed 2 @NotNull annotations as these are specific for the intellij ide and the necessary transient dependency has been removed from jitsi-utils.

See https://www.jetbrains.com/help/idea/nullable-and-notnull-annotations.html

If the annotations should stay in place, we need to add:

```
<dependency>
    <groupId>org.jetbrains</groupId>
    <artifactId>annotations</artifactId>
    <version>19.0.0</version>
</dependency>
```

Neustradamus commented:
@123Haynes: Thanks a lot for your PRs (https://github.com/jitsi/bccontrib/pull/3 and this), I am very happy :)

A long time ago, I have communicated about old dependencies on GitHub, on Twitter (private and public), and recently on the forum.

Can you look in https://github.com/jitsi/jitsi desktop client repository too?

I have done PR a long time ago (and recently) without solutions:
- https://github.com/pulls?q=is%3Apr+author%3ANeustradamus+org%3Ajitsi+

In this organization:
- Bouncycastle: https://github.com/search?q=org%3Ajitsi+bouncycastle&type=Code
- SLF4J: https://github.com/search?q=org%3Ajitsi+bouncycastle&type=Code
- ...

Neustradamus commented:
@123Haynes: I have updated the first ticket with all PRs included yours, can you look if it is good or not for other?
- https://github.com/jitsi/jitsi/issues/562

JonathanLennox commented:
For the JetBrains annotations, I think the best thing to do would be:

```
       <dependency>
           <groupId>org.jetbrains</groupId>
           <artifactId>annotations</artifactId>
           <version>15.0</version>
           <optional>true</optional>
       </dependency>
```

This allows us to use the annotation, without importing the dependency to libraries that use this one.

JonathanLennox commented:
Jenkins, test this please

123Haynes commented:
@JonathanLennox should I use the 15.0 version or the latest (19.0) ?

JonathanLennox commented:
19 is fine if it works; I just copy&paste'd this from the jitsi-utils pom.

123Haynes commented:
@JonathanLennox Thank you for your review.
I think I have addressed your comments 😃 

If you have some time, can you maybe also look at this PR?

https://github.com/jitsi/bccontrib/pull/3

I'm trying to update all dependencies starting with the most low-level libs I could find and working my way through the jitsi codebase 😄

JonathanLennox commented:
I'd rather _not_ have all the whitespace changed - it makes it hard to review the PR, and confuses the `git annotate` history.  Can you revise to just change the indentation of the new element?

JonathanLennox commented:
(Actually it looks like Jenkins isn't wired up here - I've tested it manually, and it passes its tests.)

123Haynes commented:
@JonathanLennox ok, sure. I've changed it so only the indentation of the new dependency is affected.
Note that there are several indentation errors in the pom.xml file at the moment.
I guess we just leave it like that? 🤷

JonathanLennox commented:
We can do an indentation fix, but I'd rather it be its own separate PR that only changes whitespace and nothing else.

123Haynes commented:
alright. I'll create a separate PR for the whitespace changes tomorrow.

JonathanLennox commented:
Sorry for being a moving target, but there's a newer version of jitsi-utils now.  (This is the one that updates the JNA version so you pretty definitely want it...)

123Haynes commented:
@JonathanLennox No problem, I updated it.

Neustradamus commented:
@JonathanLennox: Thanks for the merging of @123Haynes!
